x-pack/plugins/security_solution/public/management/pages/host_isolation_exceptions/view/host_isolation_exceptions_list.tsx

```diff
@@ -39,27 +38,41 @@ import {
 } from './components/translations';
 import { getEndpointListPath } from '../../../common/routing';
 import { useEndpointPrivileges } from '../../../../common/components/user_privileges/endpoint';
-import { HostIsolationExceptionsPageAction } from '../store/action';
+import { getHostIsolationExceptionItems } from '../service';
+import { parseQueryFilterToKQL } from '../../../common/utils';
 
 type HostIsolationExceptionPaginatedContent = PaginatedContentProps<
   Immutable<ExceptionListItemSchema>,
   typeof ExceptionItem
 >;
+const SEARCHABLE_FIELDS: Readonly<string[]> = [`name`, `description`, `entries.value`];
 
 export const HostIsolationExceptionsList = () => {
-  const listItems = useHostIsolationExceptionsSelector(getListItems);
-  const totalCountListItems = useHostIsolationExceptionsSelector(getTotalListItems);
-  const pagination = useHostIsolationExceptionsSelector(getListPagination);
-  const isLoading = useHostIsolationExceptionsSelector(getListIsLoading);
-  const fetchError = useHostIsolationExceptionsSelector(getListFetchError);
+  const http = useHttp();
+  const history = useHistory();
+  const privileges = useEndpointPrivileges();
+
   const location = useHostIsolationExceptionsSelector(getCurrentLocation);
-  const dispatch = useDispatch<Dispatch<HostIsolationExceptionsPageAction>>();
+  const pagination = useHostIsolationExceptionsSelector(getListPagination);
   const navigateCallback = useHostIsolationExceptionsNavigateCallback();
 
   const [itemToDelete, setItemToDelete] = useState<ExceptionListItemSchema | null>(null);
 
-  const history = useHistory();
-  const privileges = useEndpointPrivileges();
+  const { isLoading, data, error, refetch } = useQuery<
```

Comment:
I see too much code query related in this component. IMO would be better to have all this code encapsulated in a custom hook that can be reused in other places (or not, just to keep the component as much clean as possible of external code).
By query related I mean:
parseQueryFilterToKQL, getHostIsolationExceptionItems, SEARCHABLE_FIELDS, the useQuery itself, http hook

I think @paul-tavares already mentioned that, but if we want to rely on cache keys, we need to keep it encapsulated as much as possible in order to avoid key duplications and multi executions of some queries.

Thoughts on that?

Reply:
Hey, I know I was accidentally included in the review, but I would like to take the opportunity to share this [Use Query Keys factories](https://tkdodo.eu/blog/effective-react-query-keys#use-query-key-factories) pattern. It's just a simple object with entries and functions that will produce query keys, which you can then use in your custom hooks.

```js
const hostIsolationExceptionsKeys = {
  all: ['hostIsolationExceptions'] as const,
  lists: () => [...hostIsolationExceptionsKeys.all, 'list'] as const,
  list: (filters) => [...todoKeys.lists(), extractFilterParams(filters)] as const, // let the extract function here instead of omitting
}
```
then you can:

```js
const { isLoading, data, error, refetch } = useQuery(hostIsolationExceptionsKeys.list(location), () => {
// ....
}
```
and also

```js
queryClient.invalidateQueries(hostIsolationExceptionsKeys.all);
```

Reply:
Thanks for sharing I'll take a look and see if I can take advantage of that :)